recommendations/local-resource-conservation.md

```diff
@@ -0,0 +1,334 @@
+has forwarded outwards do not represent fees paid by a potential attacker.
+
+#### In Flight HTLC Risk
+Whenever a HTLC is forwarded, we assume that it will resolve with the worst 
+case off-chain resolution time and dock reputation accordingly. This decrease 
+will be temporary in the case of fast resolution, and preemptively slash 
+reputation in the case where the HTLC is used as part of a slow jamming attack.
+
+We define the following parameters: 
+* `height_added`: the block height at which the HTLC was irrevocably committed 
+  to by the local node.
+
+We define the `outstanding_risk` of in-flight HTLCs: 
+* `outstanding_risk` = `fees` * ((`cltv_expiry` - `height_added`) * 10 * 60) / `resolution_period`
+
+We define the `in_flight_htlc_risk` for an incoming channel sending 
+`update_add_htlc`:
+* `in_flight_htlc_risk` = `outstanding_risk(proposed update_add_htlc)`
+* for each HTLC originating from the incoming channel:
+  * if `endorsed` is non-zero: 
+    * `in_flight_htlc_risk` += `outstanding_risk(HTLC)`
+
+##### Rationale
+In flight HTLC are included in reputation scoring to account for sudden changes
+in a peer's behavior. Even when sufficient reputation is obtained, each HTLC 
+choosing to take advantage of that reputation is treated as if it will be used 
+to inflict maximum damage. The expiry height of each incoming in flight HTLC is 
+considered so that risk is directly related to the amount of time the HTLC 
+could be held in the channel. Ten minute blocks are assumed for simplicity.
+
+## Resource Bucketing
+When making the decision to forward a HTLC on its outgoing channel, a node 
+MAY choose to limit its exposure to HTLCs that put it at risk of a denial of 
+service attack.
+
+We define the following parameters:
+* `protected_slot_count`: defines the number of HTLC slots that are reserved 
+  for endorsed HTLCs from peers with sufficient reputation (default: 0.5 * 
+  remote peer's `max_accepted_htlcs`).
+* `protected_liquidity_portion`: defines the portion of liquidity that is 
+  reserved for endorsed HTLCs from peers with sufficient reputation (default: 
+  0.5).
+
+A node implementing resource bucketing limits exposure on its outgoing channel:
+* MUST choose `protected_slot_count` <= the remote channel peer's 
+  `max_accepted_htlcs`.
+* MUST choose `protected_liquidity_portion` in [0;1].
+
+For each `update_add_htlc` proposed by an incoming channel:
+* If `endorsed` is non-zero AND the incoming channel has sufficient local 
+  reputation for the HTLC (see [Recommendations for Reputation Scoring](#recommendations-for-reputation-scoring)):
+  * SHOULD forward the HTLC as usual.
+  * SHOULD set `endorsed` to 1 in the outgoing `update_add_htlc`.
+* Otherwise: 
+  * SHOULD reduce the remote peer's `max_accepted_htlcs` by 
+    `protected_slot_count` for the purposes of the proposed HTLC.
+  * SHOULD reduce the `max_htlc_value_in_flight` by 
+    `protected_liquidity_portion` * `max_htlc_value_in_flight`.
```

Comment:
>  * SHOULD reduce the `max_htlc_value_in_flight` by 
>    `protected_liquidity_portion` * `max_htlc_value_in_flight`.

See comment above HTLC on-the-fly upgrade from "general" to "protected". This
could be the kind of situation where a high-value HTLC paying good routing fees
is rejected from forward on outgoing channel, because the HTLC `outgoing_htlc_value` 
is falling just above `protected_liquidity_portion` * `max_htlc_value_in_flight`.

Of course, all depend if there is high volume of traffic going through
 the target node, and that traffic probabilistically should soon occupied the
`protected_slot_liquidity`, or if it's more economically interesting to take
 the risk of making an exemption.
         
One could suggest this part could be better left to be described in another
document, or a blip and have implementation experimenting with that. This
could be too "rigid" for low-volume forwarding nodes and the parameters too 
"flexible" for high-volume, topologically well-connected forwarding nodes.